lib/spack/spack/test/repository.py

```diff
@@ -0,0 +1,121 @@
+##############################################################################
+# Copyright (c) 2013-2016, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the LICENSE file for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+
+import spack
+import spack.repository
+import pytest
+
+
+@pytest.fixture()
+def mock_repository():
+    return spack.repository.RepoPath('$spack/var/spack/repos/builtin.mock')
+
+
+class TestRepoPath(object):
+
+    def test_error_during_construction(self):
+
+        # If the repository has no repo.yaml, raise BadRepoError
+        with pytest.raises(spack.repository.BadRepoError):
+            spack.repository.RepoPath('/foo/bar')
+
+    def test_magic_methods(self, mock_repository):
```

Comment:
More-descriptive name is preferred. This also appears to be testing several different behaviors; for example deleting, copying, and adding could be tested separately.